pkg/pillar/devicenetwork/dhcpcd.go

```diff
@@ -92,6 +92,17 @@ func doDhcpClientActivate(log *base.LogObject, nuc types.NetworkPortConfig) {
 		}
 		log.Functionf("dhcpcd %s not running", nuc.IfName)
 		extras := []string{"-f", "/dhcpcd.conf", "--noipv4ll", "-b", "-t", "0"}
+		switch nuc.Type {
+		case types.NtIpv4Only:
+			extras = []string{"-f", "/dhcpcd.conf", "--noipv4ll", "--ipv4only", "-b", "-t", "0"}
+		case types.NtIpv6Only:
+			extras = []string{"-f", "/dhcpcd.conf", "--noipv4ll", "--ipv6only", "-b", "-t", "0"}
```

Comment:
Presumably you don't need noipv4ll in this case.

Reply:
removed